numpy/lib/function_base.py

```diff
@@ -1897,24 +1905,21 @@ def diff(a, n=1, axis=-1):
     if n < 0:
         raise ValueError(
             "order must be non-negative but got " + repr(n))
+
     a = asanyarray(a)
     nd = a.ndim
-    slice1 = [slice(None)]*nd
-    slice2 = [slice(None)]*nd
-    slice1[axis] = slice(1, None)
-    slice2[axis] = slice(None, -1)
-    slice1 = tuple(slice1)
-    slice2 = tuple(slice2)
+    rng = range(nd)
+    axis = normalize_axis_index(axis, nd)
+    keep = slice(None)
 
-    if a.dtype == np.bool_:
-        da = a[slice1] != a[slice2]
-    else:
-        da = a[slice1] - a[slice2]
+    slice1 = tuple(slice(1, None) if i == axis else keep for i in rng)
+    slice2 = tuple(slice(None, -1) if i == axis else keep for i in rng)
```

Comment:
It also appears to be slower than the original: https://stackoverflow.com/questions/45364139/is-there-a-built-in-method-for-checking-individual-list-indices-in-python/45377475#comment77720595_45364139. I guess I'll revert it.